envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt

```diff
@@ -120,6 +121,10 @@ internal class EnvoyEgressRoutesFactory(
             routeAction.setInternalRedirectAction(RouteAction.InternalRedirectAction.HANDLE_INTERNAL_REDIRECT)
         }
 
+        routeAction
+            .setAutoHostRewrite(BoolValue.of(true))
+            .setAutoHostRewriteHeader("Custom-Host")
```

Comment:
You set `auto_host_rewrite` and `auto_host_rewrite_header`, according to [Envoy docs ](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route_components.proto.html#envoy-api-field-route-routeaction-auto-host-rewrite-header)only one should be used.

Reply:
changed